### PR TITLE
crane: ignore TLS certificate validation when using --insecure flag

### PR DESCRIPTION
**Pull request by cpanato**
when using the `--insecure` flag ignore the TLS validation.
Please let me know if this is ok or what are your thoughts

/assign @jonjohnsonjr 

Fixes: https://github.com/google/go-containerregistry/issues/1024

### CODE REVIEW

File: cmd/crane/cmd/root.go [hunk 1]
```diff
@@ -67,12 +68,26 @@ func New(use, short string, options []crane.Option) *cobra.Command {
 			cf, err := config.Load(os.Getenv("DOCKER_CONFIG"))
 			if err != nil {
 				logs.Debug.Printf("failed to read config file: %v", err)
+				return err
```
Comment (jonjohnsonjr) [inline cmd/crane/cmd/root.go:71]: We don't want this to be fatal.

File: cmd/crane/cmd/root.go [hunk 2]
```diff
@@ -67,12 +68,26 @@ func New(use, short string, options []crane.Option) *cobra.Command {
...
 			} else if len(cf.HTTPHeaders) != 0 {
 				options = append(options, crane.WithTransport(&headerTransport{
-					inner:       http.DefaultTransport,
+					inner: &http.Transport{
+						TLSClientConfig: &tls.Config{InsecureSkipVerify: insecure},
```
Comment (jonjohnsonjr) [inline cmd/crane/cmd/root.go:75]: I'd dedupe this a bit, something like:

```
transport := http.DefaultTransport.(*http.Transport).Clone()
transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: insecure}

...
} else if len(cf.HTTPHeaders) != 0 {
  transport = &headerTransport{transport, cf.HTTPHeaders}
}

options = append(options, crane.WithTransport(transport))

Comment (cpanato) [reply cmd/crane/cmd/root.go]: I applied some changes if I try your suggestion or something related I get this error
`cannot use &(headerTransport literal) (value of type *headerTransport) as *http.Transport value in assignment`

with my limited knowledge, I don't know how to solve it.

Comment (cpanato) [reply cmd/crane/cmd/root.go]: thanks for your review and feedback

File: cmd/crane/cmd/root.go [hunk 3]
```diff
@@ -64,30 +64,23 @@ func New(use, short string, options []crane.Option) *cobra.Command {
...
-				return nil
-			}
 
-			if insecure {
-				options = append(options, crane.WithTransport(&headerTransport{
-					inner: &http.Transport{
-						TLSClientConfig: &tls.Config{InsecureSkipVerify: insecure},
-					},
-				}))
+				return
```
Comment (jonjohnsonjr) [inline cmd/crane/cmd/root.go:80]: Drop this return, otherwise lgtm.

File: cmd/crane/cmd/root.go (hunk 2 above)
Comment (jonjohnsonjr) [reply cmd/crane/cmd/root.go]: > cannot use &(headerTransport literal) (value of type *headerTransport) as *http.Transport value in assignment

Ah, yeah i was just using some shorthand, what you did instead works 👍

File: cmd/crane/cmd/root.go (hunk 3 above)
Comment (cpanato) [reply cmd/crane/cmd/root.go:80]: done, thanks again! PTAL

**Review by jonjohnsonjr — APPROVED** [review]
Thanks!

**Pull request merged**